### PR TITLE
adding BugzillaReportsTool under tools

### PR DESCRIPTION
**Pull request by hmaheswa**
Signed-off-by: Hemanth Maheswarla <hmaheswa@hmaheswa.remote.csb>

### CODE REVIEW

**Comment by neha-gangadhar** [comment]
Hi Pragadeeswaran, 

Could you please help review the PR.

Thanks,
Neha

**Comment by neha-gangadhar** [comment]
Hi Hemanth Sai, 

Many files in the code repo pushed are not being used. Can we delete them if not needed anymore to have the repo clean?

Thanks,
Neha

**Comment by psathyan** [comment]
@hmaheswa the changes are large and would prefer adding more information to the PR description

Could you fix the failing tox?

File: tools/bugzilla-reports-tool/Main.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8
```
Comment (psathyan) [inline tools/bugzilla-reports-tool/Main.py:1]: Not important, but would prefer to use all lower cases for the files

File: tools/bugzilla-reports-tool/Main.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import smtplib
+import sys
+import time
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+
+sys.path.append("cephQeInfra")
```
Comment (psathyan) [inline tools/bugzilla-reports-tool/Main.py:8]: Could we switch to all lower cases?

File: tools/bugzilla-reports-tool/Main.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8
+import smtplib
```
Comment (psathyan) [inline tools/bugzilla-reports-tool/Main.py:2]: Please add a description on the purpose of this module and how to execute

File: tools/bugzilla-reports-tool/Main.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+import time
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+
+sys.path.append("cephQeInfra")
+from datetime import datetime
+
+import pytz
+from cephQeInfra import (
+    docOnQa,
```
Comment (psathyan) [inline tools/bugzilla-reports-tool/Main.py:13]: Please switch all module names to lower case.

File: tools/bugzilla-reports-tool/Main.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    s.quit()
+    print(
+        "Results have been emailed to {recipients}".format(
+         recipients=recipients
+         )
+        )
+
+except Exception as e:
+            print("\n")
+            log.exception(e)
```
Comment (psathyan) [inline tools/bugzilla-reports-tool/Main.py:87]: Could we indent correctly?

Also there is not `logging` module. IMO, we should have `stdout` and `stderr` streams configured for enhancing our logging.

File: tools/bugzilla-reports-tool/.gitignore [hunk 6]
```diff
@@ -0,0 +1,105 @@
+# Byte-compiled / optimized / DLL files
```
Comment (psathyan) [inline tools/bugzilla-reports-tool/.gitignore:1]: Could we remove this file? As the root `.gitignore` would handle most cases. If they are handled, please add it to the `.gitignore` that is present at root level.

File: tools/bugzilla-reports-tool/README.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
+# bugzilla-reports-tool
+
+This tool will allow you to create a Quality dashboard for your product and send (weekly or whenever you want, depend on cronjob) reports with bug status.
+The quality dashboard is fully automated and the template Google Spreadsheet can be found here - https://docs.google.com/spreadsheets/d/1K2ImPbbFO52njNC2CjvRT5zpzO7wcvOp5L3GSkkGoR8/edit?usp=sharing
+
+## Quality dashboard
+An exeample of this dashboard:
+
+![alt text](https://github.com/RazTamir/bugzilla-reports-tool/blob/master/pics/Screenshot%20from%202020-04-05%2015-23-43.png)
```
Comment (psathyan) [inline tools/bugzilla-reports-tool/README.md:9]: Fix the hyper link

**Comment by hmaheswa** [comment]
Sure Neha, I will remove the unused code

File: tools/bugzilla-reports-tool/README.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+10. Execute with:
+
+$ python3.7 release_readiness_criteria.py SPREADSHEET_NAME
+
+11. A list of recommended cronjob exist in the 'Dashboard configuration' tab
+
+## QE Tracking dashboard
+An exeample of this dashboard:
+
+![alt text](https://github.com/RazTamir/bugzilla-reports-tool/blob/master/pics/Screenshot%20from%202019-10-28%2012-46-50.png)
```
Comment (psathyan) [inline tools/bugzilla-reports-tool/README.md:48]: Fix the hyper links

File: tools/bugzilla-reports-tool/README.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+### Usage:
+Same steps as above with changing the script you execute in step 8 to:
+
+$ python3.7 tracking_dashboard.py SPREADSHEET_NAME
+
+## Arrival vs. Resolved trend
+An exeample of this dashboard:
+
+![alt text](https://github.com/RazTamir/bugzilla-reports-tool/blob/master/pics/Screenshot%20from%202019-11-18%2018-04-00.png)
```
Comment (psathyan) [inline tools/bugzilla-reports-tool/README.md:57]: Fix the hyper links

**Comment by hmaheswa** [comment]
okay Pragadeesh, will fix tox errors and add info to PR description

File: tools/bugzilla-reports-tool/arrivals_vs_resolved.py [hunk 10]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python
+from datetime import datetime
+
+from helpers import *
```
Comment (psathyan) [inline tools/bugzilla-reports-tool/arrivals_vs_resolved.py:4]: It is a good practice to avoid blind imports. Please refactor the code to avoid it.

**Review by psathyan — COMMENTED** [review]
Part I of my comments.

**Comment by psathyan** [comment]
Thanks for pulling this together. As the number of lines are large, I prefer doing the review in pieces so that we adhere to the contribution guidelines.

**Comment by hmaheswa** [comment]
closing this PR as this code is added in the Gitlab repo https://gitlab.cee.redhat.com/rhcs-qe/reporter-bugzilla/-/tree/main

**Pull request closed without merging**